File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.js
/**
 * @param {string} s
 * @param {string} part
 * @return {string}
 */
var removeOccurrences = function(s, part) {
    while (s.indexOf(part) !== -1) {
        let idx = s.indexOf(part);
        s = s.slice(0, idx) + s.slice(idx + part.length);
    }
    return s;
};

File: 2021-remove-all-occurrences-of-a-substring/remove-all-occurrences-of-a-substring.py
class Solution:
    def removeOccurrences(self, s: str, part: str) -> str:
        s_list = s.split()
        partLen = len(part)

        while part in s:
            idx = s.find(part)
            s = s[:idx] + s[(idx + partLen):]
        
        return s